include/nuttx/sensors/qmi8658.h

```diff
@@ -0,0 +1,310 @@
+#define QMI8658_ACC_ODR_1000Hz         (0x03)
+#define QMI8658_ACC_ODR_500Hz          (0x04)
+#define QMI8658_ACC_ODR_250Hz          (0x05)
+#define QMI8658_ACC_ODR_125Hz          (0x06)
+#define QMI8658_ACC_ODR_62_5Hz         (0x07)
+#define QMI8658_ACC_ODR_31_25Hz        (0x08)
+#define QMI8658_ACC_ODR_LOWPOWER_128Hz (0x0C)
+#define QMI8658_ACC_ODR_LOWPOWER_21Hz  (0x0D)
+#define QMI8658_ACC_ODR_LOWPOWER_11Hz  (0x0E)
+#define QMI8658_ACC_ODR_LOWPOWER_3Hz   (0x0F)
+
+/* Gyroscope ODR */
+#define QMI8658_GYRO_ODR_7174_4Hz  (0x00)
+#define QMI8658_GYRO_ODR_3587_2Hz  (0x01)
+#define QMI8658_GYRO_ODR_1793_6Hz  (0x02)
+#define QMI8658_GYRO_ODR_896_8Hz   (0x03)
+#define QMI8658_GYRO_ODR_448_4Hz   (0x04)
+#define QMI8658_GYRO_ODR_224_2Hz   (0x05)
+#define QMI8658_GYRO_ODR_112_1Hz   (0x06)
+#define QMI8658_GYRO_ODR_56_05Hz   (0x07)
+#define QMI8658_GYRO_ODR_28_025Hz  (0x08)
+
+/* Low-Pass Filter Modes */
+#define QMI8658_LPF_MODE_0         (0x00)
+#define QMI8658_LPF_MODE_1         (0x01)
+#define QMI8658_LPF_MODE_2         (0x02)
+#define QMI8658_LPF_MODE_3         (0x03)
+#define QMI8658_LPF_OFF            (0x04)
+
+/* Sample Synchronization Modes */
+#define QMI8658_SYNC_MODE          (0x00)
+#define QMI8658_ASYNC_MODE         (0x01)
+
+/****************************************************************************
+ * IOCTL Commands
+ ****************************************************************************/
+
+#define SNIOC_SET_ACC_RANGE  _SNIOC(0x1)
+#define SNIOC_SET_GYRO_RANGE _SNIOC(0x2)
+#define SNIOC_ENABLE_SENSOR  _SNIOC(0x3)
+#define SNIOC_DISABLE_SENSOR _SNIOC(0x4)
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+struct qmi8658_data_s
+{
+  int16_t accel_x;
+  int16_t accel_y;
+  int16_t accel_z;
+  int16_t gyro_x;
+  int16_t gyro_y;
+  int16_t gyro_z;
+  int16_t temperature;
+};
+
+/* Configuration and scale factor structures are defined in qmi8658_base.h
+ * for internal driver use
+ */
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/****************************************************************************
+ * Name: qmi8658_register
+ *
+ * Description:
+ *   Register the QMI8658 character device at 'devpath'
+ *
+ * Input Parameters:
+ *   devpath - The full path to the driver to register. E.g., "/dev/imu0"
+ *   i2c     - Pointer to the I2C master device
+ *   addr    - I2C address (use QMI8658_I2C_ADDR for default)
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int qmi8658_register(FAR const char *devpath,
+                     FAR struct i2c_master_s *i2c, uint8_t addr);
```

Comment:
NuttX is moving away from the legacy sensor driver interface (char device). I don't think you should support a second version in addition to uORB, it makes more work to maintain.